src/backend/distributed/commands/utility_hook.c

```diff
@@ -332,10 +332,10 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 			ResetConstraintDropped();
 
 			/*
-			 * We're only interested in top-level CREATE TABLE commands
+			 * We're not interested in CREATE TABLE commands in the form a subcommand
 			 * to create a tenant schema table or a Citus managed table.
 			 */
-			if (context == PROCESS_UTILITY_TOPLEVEL &&
+			if (context != PROCESS_UTILITY_SUBCOMMAND &&
```

Comment:
This is also how we decide converting a local table to a Citus local table when defining a foreign key to a reference table.